mwclient/client.py

```diff
@@ -440,13 +530,38 @@ def raw_api(self, action, http_method='POST', *args, **kwargs):
             raise errors.InvalidResponse(res)
 
     def raw_index(self, action, http_method='POST', *args, **kwargs):
-        """Sends a call to index.php rather than the API."""
+        """Sends a call to index.php rather than the API.
+
+        Args:
+            action (str): The MediaWiki API action to perform.
+            http_method (str): The HTTP method to use in the request.
+
+        Returns:
+            The API response.
+        """
         kwargs['action'] = action
         kwargs['maxlag'] = self.max_lag
         data = self._query_string(*args, **kwargs)
         return self.raw_call('index', data, http_method=http_method)
 
     def require(self, major, minor, revision=None, raise_error=True):
+        """Check whether the current wiki matches the required version.
+
+        Args:
+            major (int): The required major version.
+            minor (int): The required minor version.
+            revision (int): The required revision.
+            raise_error (bool): Whether to throw an error if the version of the current
+                wiki is below the required version. Defaults to `True`.
+
+        Returns:
+            `False` if the version of the current wiki is below the required version, else
+                `True`. If `raise_error=True` then nothing is returned.
+
+        Raises:
+            errors.MediaWikiVersionError: The thw current wiki is below the required
```

Comment:
"thw" here seems like a typoed duplicate of "The". Should be omitted.

This can also raise `NotImplementedError` if the `revision` arg is set. Which is...weird. Why does it do that? Bit of a footgun.

Reply:
It also raises `RuntimeError` if the site is not initialized and `raise_error` is anything but `None`.

Reply:
I've added the `NotImplementedError` and `RuntimeError` to the docstring, this method is definitely a bit weird :)

```
RuntimeError: It `raise_error` is `None` and the `version` attribute is unset
    This is usually done automatically on construction of the :class:`Site`,
    unless `do_init=False` is passed to the constructor. After instantiation,
    the :meth:`~Site.site_init` functon can be used to retrieve and set the
    `version`.
NotImplementedError: If the `revision` argument was passed. The logic for this
    is currently unimplemented.
```